--- application.sb~	2013-10-24 18:47:45.000000000 -0700
+++ application.sb	2013-10-24 18:52:04.000000000 -0700
@@ -952,11 +952,20 @@
         (entitlement "com.apple.security.network.server"))
   (begin
     (allow mach-lookup (global-name "org.h5l.kcm"))
+    (allow file-write*
+           (home-regex #"/Library/Caches/com.padl.gss.BrowserID/browserid.tickets.json*"))
     (allow file-read*
+           (subpath "/usr/lib/sasl2")
+           (subpath "/usr/local/etc")
+           (subpath "/usr/local/lib")
+           (subpath "/usr/local/share")
+           (literal "/private/etc/krb5.conf")
+           (literal "/private/etc/gss/mech")
            (literal "/Library/Preferences/edu.mit.Kerberos")
            (literal "/Library/Preferences/com.apple.Kerberos.plist")
            (literal "/private/etc/krb5.conf")
            (regex "^/Library/Preferences/com\\.apple\\.GSS\\..*\\.plist$")
+           (home-regex #"/Library/Caches/com.padl.gss.BrowserID/browserid.tickets.json*")
            (home-literal "/Library/Preferences/edu.mit.Kerberos")
            (home-literal "/Library/Preferences/com.apple.Kerberos.plist"))))
 (allow file-read*
